src/client/utils/dom.js

```diff
@@ -8,15 +8,19 @@ import { sameOriginCheck } from './destination-location';
 import { isFirefox, isWebKit, isIE, version as browserVersion } from './browser';
 import trim from '../../utils/string-trim';
 import getNativeQuerySelectorAll from './get-native-query-selector-all';
-
-const NATIVE_METHOD_REG_EX = /^\s*function\s[\s\S]+()\s*\{\s*\[native\scode\]\s*\}\s*$/;
+import { instanceAndPrototypeToStringAreEqual } from '../utils/feature-detection';
 
 // NOTE: We should avoid using native object prototype methods,
 // since they can be overriden by the client code. (GH-245)
-var arraySlice = Array.prototype.slice;
+const arraySlice = Array.prototype.slice;
 
 var scrollbarSize = null;
 
+const IS_NOT_DOM_ELEMENT_RE = /^\[object (Element|Node|EventTarget)(Prototype)?]$/;
+const IS_WINDOW_RE          = /^\[object (Window|global)]$/;
+const IS_DOCUMENT_RE        = /^\[object (HTML)?Document]$/;
+const IS_SVG_ELEMENT_RE     = /^\[object SVG\w+?Element]$/i;
+
```

Comment:
`indexOf` is faster then regex. https://jsperf.com/regex-versus-indexofwithtwoelements2
use  `['[object Window]', '[object global]']`.indexOf` instead of `IS_WINDOW_RE`.